tools/codegen/model.py

```diff
@@ -1496,3 +1509,66 @@ def parse_returns(return_decl: str) -> Tuple[Return, ...]:
     if return_decl[0] == '(' and return_decl[-1] == ')':
         return_decl = return_decl[1:-1]
     return tuple(Return.parse(arg) for arg in return_decl.split(', '))
+
+# A PrecomputedElement consists of the name of such an element and its C++ type.
+# Together with `Precompute` below, it helps capture all information about
+# precomputed elements needed to properly generate code for a kernel that has them.
+@dataclass(frozen=True)
+class PrecomputedElement:
+    # The name of the precomputed element.
+    name: str
+    # The C++ type of the precomputed element.
+    cpp_ty: str
```

Comment:
since you switched these to JIT type, this should get updated too right?

Reply:
Yep